File: c/4/6064.c
#include <stdio.h>

int gcd(int a, int b) {
    if (a < b) {
        int tmp = a;
        a = b;
        b = tmp;
    }

    while (b > 0) {
        int tmp = a;
        a = b;
        b = tmp%b;
    }

    return a;
}

void tick() {
    // get m, n, x, y
    int m, n, x, y;
    scanf("%d %d %d %d", &m, &n, &x, &y);

    // check validity
    int g = gcd(m, n);
    if (x > y && (x-y) % g != 0 || x < y && (y-x) % g != 0) {
        printf("-1\n");
        return;
    }

    // calculate year
    while (x != y) {
        if (x < y) {
            x += m;
        } else {
            y += n;
        }
    }

    // print result
    printf("%d\n", x);
}

int main() {
    // get testcase count
    int testcase_count;
    scanf("%d", &testcase_count);

    // repeat `count` times
    for (int i = 0; i < testcase_count; i++) {
        tick();
    }

    return 0;
}

File: rust/src/4/6064.rs
use std::io::stdin;

fn gcd(mut a: u32, mut b: u32) -> u32 {
    if a < b {
        (a, b) = (b, a);
    }

    while b > 0 {
        (a, b) = (b, a%b);
    }

    a
}

fn tick() {
    // get m, n, x, y
    let mut input = String::new();
    stdin().read_line(&mut input).unwrap();
    let mut iter = input
        .trim()
        .split_whitespace()
        .map(|x| x.parse::<u32>().unwrap());
    let m = iter.next().unwrap();
    let n = iter.next().unwrap();
    let mut x = iter.next().unwrap();
    let mut y = iter.next().unwrap();

    // check validality
    let g = gcd(m, n);
    if x > y && (x-y) % g != 0
            || x < y && (y-x) % g != 0 {
        println!("-1");
        return;
    }

    // calculate year
    while x != y {
        if x < y {
            x += m;
        } else {
            y += n;
        }
    }

    // print result
    println!("{}", x);
}

fn main() {
    // get testcase count
    let mut input = String::new();
    stdin().read_line(&mut input).unwrap();
    let testcase_count = input.trim().parse::<u32>().unwrap();

    // repeat `count` times
    for _ in 0..testcase_count {
        tick();
    }
}
